homeassistant/components/litterrobot/switch.py

```diff
@@ -1,76 +1,89 @@
 
-from pylitterbot import LitterRobot
+from pylitterbot import FeederRobot, LitterRobot
 
-from homeassistant.components.switch import SwitchEntity
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .const import DOMAIN
-from .entity import LitterRobotConfigEntity
+from .entity import LitterRobotConfigEntity, _RobotT
 from .hub import LitterRobotHub
 
 
-class LitterRobotNightLightModeSwitch(
-    LitterRobotConfigEntity[LitterRobot], SwitchEntity
-):
-    """Litter-Robot Night Light Mode Switch."""
+@dataclass
+class RequiredKeysMixin(Generic[_RobotT]):
+    """A class that describes robot switch entity required keys."""
 
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if switch is on."""
-        if self._refresh_callback is not None:
-            return self._assumed_state
-        return self.robot.night_light_mode_enabled
+    icons: tuple[str, str]
+    set_fn: Callable[[_RobotT], Callable[[bool], Coroutine[Any, Any, bool]]]
 
-    @property
-    def icon(self) -> str:
-        """Return the icon."""
-        return "mdi:lightbulb-on" if self.is_on else "mdi:lightbulb-off"
 
-    async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the switch on."""
-        await self.perform_action_and_assume_state(self.robot.set_night_light, True)
+@dataclass
+class RobotSwitchEntityDescription(SwitchEntityDescription, RequiredKeysMixin[_RobotT]):
+    """A class that describes robot switch entities."""
+
+
+ROBOT_SWITCHES = [
+    RobotSwitchEntityDescription[Union[LitterRobot, FeederRobot]](
+        key="night_light_mode_enabled",
+        name="Night Light Mode",
+        icons=("mdi:lightbulb-on", "mdi:lightbulb-off"),
+        set_fn=lambda robot: robot.set_night_light,
+    ),
+    RobotSwitchEntityDescription[Union[LitterRobot, FeederRobot]](
+        key="panel_lock_enabled",
+        name="Panel Lockout",
+        icons=("mdi:lock", "mdi:lock-open"),
+        set_fn=lambda robot: robot.set_panel_lockout,
+    ),
+]
 
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the switch off."""
-        await self.perform_action_and_assume_state(self.robot.set_night_light, False)
 
+class RobotSwitchEntity(LitterRobotConfigEntity[_RobotT], SwitchEntity):
+    """Litter-Robot switch entity."""
 
-class LitterRobotPanelLockoutSwitch(LitterRobotConfigEntity[LitterRobot], SwitchEntity):
-    """Litter-Robot Panel Lockout Switch."""
+    entity_description: RobotSwitchEntityDescription[_RobotT]
+
+    def __init__(
+        self,
+        robot: _RobotT,
+        hub: LitterRobotHub,
+        description: RobotSwitchEntityDescription[_RobotT],
+    ) -> None:
+        """Initialize a Litter-Robot switch entity."""
+        assert description.name
```

Comment:
Ideally, if possible it would be nice to get rid of this assert in a future PR.

Reply:
I did this in https://github.com/home-assistant/core/pull/77605